### ISSUE TITLE
Changing cron schedule  for recurring task

### ISSUE THREAD

**Issue by LeslieMurphy** [kagkarlsson/db-scheduler#192]
Is it possible to change a cron based task schedule dynamically?  Or possible to delete the existing task and replace it with a new one with the updated schedule?

**Comment by kagkarlsson** [comment]
Yes, if the next execution-time for a _deterministic_  `Schedule` is different from the current scheduled next execution it will be updated to reflect the modified `Schedule`.

You should see this log-line here:

https://github.com/kagkarlsson/db-scheduler/blob/master/db-scheduler/src/main/java/com/github/kagkarlsson/scheduler/task/helper/ScheduleRecurringOnStartup.java#L79

**Comment by kagkarlsson** [comment]
My goal for db-scheduler is to support a mode of operation where you rather than modifying existing records can achieve the same effect by deploying a new version of your app with changes to scheduler/task config

**Comment by LeslieMurphy** [comment]
I am hoping there is a way to have my application stay up and running, and dynamically be able to change the cron schedule on the existing task via a API call into my controller.  So instead of the cron schedule being predetermined at application startup, it could be altered during execution.

Also see my related comment in issue #193.  

Is there some way to accomplish this? 

I tried code to remove the cron task and then add in a new recurring task. I used a FixedDelay for the new task as an experiment. That approach did not work.

A DB row is created for TASK_NAME **my-new-task** TASK_INSTANCE **NewRecurring** however the scheduler logs errors.


```
import com.github.kagkarlsson.scheduler.task.TaskInstanceId;
import com.github.kagkarlsson.scheduler.Scheduler;
import com.github.kagkarlsson.scheduler.task.helper.Tasks;
import com.github.kagkarlsson.scheduler.task.helper.RecurringTask;

@Autowired
Scheduler dbScheduler;  // Scheduler implements SchedulerClient API


public void updateRecurringTask() {
 // remove existing cron task and replace it with a new recurring task
TaskInstanceId taskInstanceId = TaskInstanceId.of("my-cron-taskBEAN", "recurring");
var myTask = dbScheduler.getScheduledExecution(taskInstanceId);
dbScheduler.cancel(taskInstanceId);


// try to schedule a new recurring

RecurringTask<Void> everyMinuteTask = Tasks.recurring("my-new-task", FixedDelay.ofMinutes(1))
                    .execute((inst, ctx) -> {
                        System.out.println("Executed new minute task!");
                    });

dbScheduler.schedule(
        everyMinuteTask.instance("NewRecurring"),
        Instant.now());

// expected:  the new task starts running every minute

// actual: 

// c.g.kagkarlsson.scheduler.TaskResolver   : Found execution with unknown task-name 'my-new-task'. Adding it to the list of known unresolved task-names.
// c.g.k.scheduler.jdbc.JdbcTaskRepository  : Failed to find implementation for task with name 'my-new-task'. Execution will be excluded from due. Either delete the execution from the database, or add an implementation for it. The scheduler may be configured to automatically delete unresolved tasks after a certain period of time.


```

**Comment by kagkarlsson** [comment]
See comment here: https://github.com/kagkarlsson/db-scheduler/issues/193#issuecomment-814144634

This is a different type of RecurringTask, one with persistent Schedule. Should be possible to create, but I do not have any examples of it yet. Interesting use-case though 👍 . Will try and add an example for it

**Comment by kagkarlsson** [comment]
Threw together an example, what do you think about this approach?

https://github.com/kagkarlsson/db-scheduler/blob/persistent_schedule/examples/features/src/main/java/com/github/kagkarlsson/examples/PersistentDynamicScheduleMain.java#L33

**Comment by kagkarlsson** [comment]
🎉 This issue has been resolved in `v16.9.0` ([Release Notes](https://github.com/kagkarlsson/db-scheduler/releases/tag/v16.9.0))
